### ISSUE TITLE
Check validity of property-of-property

### ISSUE THREAD

**Issue by kzangeli** [Fiware/context.Orion-LD#307]
Second level attributes aren't check for validity - must be a JSON object with a type that is either "Property", "Relationship", or "GeoProperty", etc.

Exceptions are:
* observedAt (DateTime string)
* unitCode (string)
* datasetId (URI string)
* createdAt (ignored DateTime string)
* modifiedAt (ignored DateTime string)

**Comment by jason-fox** [comment]
Example - this is invalid

```console
curl -X POST \
  'http://localhost:1026/ngsi-ld/v1/entities/' \
  -H 'Content-Type: application/ld+json' \
  -d '{
  "id": "urn:ngsi-ld:Product:002",
  "type": "Product",
  "price": {"type": "Property","value": 10.99,"currency": "EUR"},
  "size": {"type": "Property","value": "M"},
  "name": {"type": "Property","value": "Red Wine"},
    "@context": "https://fiware.github.io/tutorials.Step-by-Step/tutorials-context.jsonld"
}'
```

It should be

```console
curl -X POST \
  'http://localhost:1026/ngsi-ld/v1/entities/' \
  -H 'Content-Type: application/ld+json' \
  -d '{
  "id": "urn:ngsi-ld:Product:002",
  "type": "Product",roperty","value": 0.99,"currency": {"type": "Property", "value": "EUR"}},
  "size": {"type": "Property","value": "S"},
  "name": {"type": "Property","value": "Gin"},
    "@context":   "https://fiware.github.io/tutorials.Step-by-Step/tutorials-context.jsonld"
}'
```

String values are only allowed for  only observedAt and unitCode , datasetId,  createdAt and modifiedAt

**Comment by jason-fox** [comment]
Related: 

```console
curl -X POST   'http://localhost:9090/ngsi-ld/v1/entities/'   -H 'Content-Type: application/ld+json'   -d '{
  "id": "urn:ngsi-ld:Product:017",
  "type": "Product",
  "price": {"type": "Property","value": 0.99,
    "currency": {"type": "Property", "value": "EUR"},
      "observedAt": "2017-07-29T12:00:04Z",
      "unitCode": "",
      "datasetId": ""
  },
  "size": {"type": "Property","value": "S"},
  "name": {"type": "Property","value": "Gin"},
 
 "@context":   "https://fiware.github.io/tutorials.Step-by-Step/tutorials-context.jsonld"
}'
```

is a valid request but:

```console
curl -X POST   'http://localhost:9090/ngsi-ld/v1/entities/'   -H 'Content-Type: application/ld+json'   -d '{
  "id": "urn:ngsi-ld:Product:017",
  "type": "Product",
  "price": {"type": "Property","value": 0.99,
    "currency": {"type": "Property", "value": "EUR"},
      "observedAt": "",
      "unitCode": "",
      "datasetId": ""
  },
  "size": {"type": "Property","value": "S"},
  "name": {"type": "Property","value": "Gin"},
 
 "@context":   "https://fiware.github.io/tutorials.Step-by-Step/tutorials-context.jsonld"
}'
```

Should return a **BadRequestData**



> 5.5.4 JSON-LD validation
> All the operations that take a JSON-LD document as input shall process such JSON-LD document as
> follows:
> • If the request payload body is not a valid JSON document then an error of type InvalidRequest shall be 
> raised.
> • If the data included by the JSON-LD document is not syntactically correct, according to the @context
> or the API Data type definitions, then an error of type BadRequestData shall be raised. For the avoidance of doubt,
> this includes validation of member values in accordance with their data type, for instance DateTime, and
> validation of each Relationship's target object which shall be a syntactically valid URI.

**Comment by MohamedSadiq102** [comment]
Broker now works

**Comment by MohamedSadiq102** [comment]
fixed close #307
